plugins/temporal/rpc.go

```diff
@@ -0,0 +1,492 @@
+	// Optional: defaulted to 10 years.
+	WorkflowExecutionTimeout time.Duration `json:"workflow_execution_timeout"`
+
+	// WorkflowRunTimeout - The timeout for duration of a single workflow run.
+	// The resolution is seconds.
+	// Optional: defaulted to WorkflowExecutionTimeout.
+	WorkflowRunTimeout time.Duration `json:"workflow_run_timeout"`
+
+	// WorkflowTaskTimeout - The timeout for processing workflow task from the time the worker
+	// pulled this task. If a workflow task is lost, it is retried after this timeout.
+	// The resolution is seconds.
+	// Optional: defaulted to 10 secs.
+	WorkflowTaskTimeout time.Duration `json:"workflow_task_timeout"`
+}
+
+/*
+- the method's type is exported.
+- the method is exported.
+- the method has two arguments, both exported (or builtin) types.
+- the method's second argument is a pointer.
+- the method has return type error.
+*/
+type Rpc struct {
+	client client.Client
+}
+
+func (p *Provider) Name() string {
+	return name
+}
+
+//func (t *T) MethodName(argType T1, replyType *T2) error
+func (p *Provider) RpcService() (interface{}, error) {
+	c, err := p.GetClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Rpc{
+		client: c,
+	}, nil
+}
+
+// ExecuteWorkflow starts a workflow execution and return a WorkflowRun instance and error
+// The user can use this to start using a function or workflow type name.
+// Either by
+//     ExecuteWorkflow(ctx, options, "workflowTypeName", arg1, arg2, arg3)
+//     or
+//     ExecuteWorkflow(ctx, options, workflowExecuteFn, arg1, arg2, arg3)
+// The errors it can return:
+//	- EntityNotExistsError, if namespace does not exists
+//	- BadRequestError
+//	- InternalServiceError
+//
+// WorkflowRun has 3 methods:
+//  - GetWorkflowID() string: which return the started workflow ID
+//  - GetRunID() string: which return the first started workflow run ID (please see below)
+//  - Get(ctx context.Context, valuePtr interface{}) error: which will fill the workflow
+//    execution result to valuePtr, if workflow execution is a success, or return corresponding
+//    error. This is a blocking API.
+// NOTE: if the started workflow return ContinueAsNewError during the workflow execution, the
+// return result of GetRunID() will be the started workflow run ID, not the new run ID caused by ContinueAsNewError,
+// however, Get(ctx context.Context, valuePtr interface{}) will return result from the run which did not return ContinueAsNewError.
+// Say ExecuteWorkflow started a workflow, in its first run, has run ID "run ID 1", and returned ContinueAsNewError,
+// the second run has run ID "run ID 2" and return some result other than ContinueAsNewError:
+// GetRunID() will always return "run ID 1" and  Get(ctx context.Context, valuePtr interface{}) will return the result of second run.
+// NOTE: DO NOT USE THIS API INSIDE A WORKFLOW, USE workflow.ExecuteChildWorkflow instead
+
+type ExecuteWorkflowIn struct {
+	Options client.StartWorkflowOptions `json:"options"`
```

Comment:
i wonder if we really need to define name mapping since we still use native structures

Reply:
i mean it's not going to be consistent

Reply:
This is an issue. Should be our StartWorkflowOptions with our JSON tags. Fixed.